kliff/dataset/weight.py

```diff
@@ -0,0 +1,167 @@
+from typing import Any, Dict, Optional
+import copy
+
+import numpy as np
+
+
+class Weight:
+    """
+    Base class for weight.
+
+    This class mimics the behavior  provided by the ``residual_data`` argument from the
+    previous version.
+    Subclass can implement
+        - compute_weight
+
+    Args:
+        config_weight: the weight for the entire configuration
+        energy_weight: the weight for the energy
+        forces_weight: the weight for the forces
+        stress_weight: the weight for the stress
+    """
+
+    def __init__(
+        self,
+        config_weight: float = 1.0,
+        energy_weight: float = 1.0,
+        forces_weight: float = 1.0,
+        stress_weight: float = 1.0,
+    ):
+        self._config_weight = config_weight  # weight for the entire configuration
+        self._energy_weight = energy_weight
+        self._forces_weight = forces_weight
+        self._stress_weight = stress_weight
+
+    @staticmethod
+    def compute_weight(config):
+        pass
+
+    @property
+    def config_weight(self):
+        return self._config_weight
+
+    @property
+    def energy_weight(self):
+        return self._energy_weight
+
+    @property
+    def forces_weight(self):
+        return self._forces_weight
+
+    @property
+    def stress_weight(self):
+        return self._stress_weight
+
+
+default_weight_params = {
+    "energy_weight_params": [1.0, 0.0],
+    "forces_weight_params": [1.0, 0.0],
+    "stress_weight_params": [1.0, 0.0],
+}
+
+
+class NonuniformWeight(Weight):
+    r"""
+    Non-uniform weight that is computed from the data. The calculation follows Lenosky et
```

Comment:
Would be good to give a full reference at the end of the doc string.

Reply:
I will do this. What do you think about the name of the class? I actually think that `NonuniformWeight` is not very descriptive. Do you have some suggestion for the name?

Reply:
Maybe `MagnitudeInverseWeight`, indicating that the weight is proportional to the inverse of the magnitude?